checkers/typeDefFirst_checker.go

```diff
@@ -0,0 +1,75 @@
+package checkers
+
+import (
+	"go/ast"
+	"go/token"
+	"strings"
+
+	"github.com/go-critic/go-critic/checkers/internal/astwalk"
+	"github.com/go-critic/go-critic/framework/linter"
+)
+
+func init() {
+	var info linter.CheckerInfo
+	info.Name = "typeDefFirst"
+	info.Tags = []string{"style", "experimental"}
+	info.Summary = "File-scoped checker, that requires type definition before its method definitions"
```

Comment:
`File-scoped checker` I'm curious - should be use this phrase in other checkers? maybe another tag or subtype? @quasilyte :)